File: mobile_app/lib/models/listing.dart
class Listing {
  final String id;
  final String title;
  final String city;
  final String category;
  final String suitableFor;
  final int monthlyRent;

  Listing({
    required this.id,
    required this.title,
    required this.city,
    required this.category,
    required this.suitableFor,
    required this.monthlyRent,
  });

  factory Listing.fromJson(Map<String, dynamic> json) {
    return Listing(
      id: json['_id'] as String,
      title: json['title'] as String,
      city: json['city'] as String,
      category: json['category'] as String,
      suitableFor: json['suitableFor'] as String,
      monthlyRent: json['monthlyRent'] as int,
    );
  }
}

File: backend/src/models/Listing.js
import mongoose from 'mongoose';

const listingSchema = new mongoose.Schema(
  {
    title: { type: String, required: true },
    description: { type: String },
    city: { type: String, required: true },
    address: { type: String, required: true },
    monthlyRent: { type: Number, required: true },
    category: {
      type: String,
      enum: ['room', 'flat', 'pg', 'hostel'],
      default: 'room',
    },
    suitableFor: {
      type: String,
      enum: ['student', 'family', 'any'],
      default: 'any',
    },
    ownerId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    isActive: { type: Boolean, default: true },
  },
  { timestamps: true }
);

export const Listing = mongoose.model('Listing', listingSchema);
